src/components/tooltip.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import Tippy, { TippyProps } from '@tippyjs/react';
+import { Box, BoxProps } from '@stacks/ui';
+import { memo, useMemo } from 'react';
+
+export const Tooltip: React.FC<
+  TippyProps & { label?: TippyProps['content']; labelProps?: BoxProps }
+> = memo(({ label, labelProps = {}, children, ...rest }) => {
+  if (!label) return <>{children}</>;
+  const content = useMemo(
```

Comment:
Do we really need to double-memo? IMO it just hurts readability and I can't imagine it gains anything, since we have the outer memo.

Reply:
it does reduce un-needed re-renders. it would be equivalent to this:

```tsx
import * as React from "react";
import Tippy, { TippyProps } from "@tippyjs/react";
import { Box, BoxProps } from "@stacks/ui";
import { memo } from "react";

const Content: React.FC<{ label: TippyProps["content"] } & BoxProps> = memo(
  ({ label, ...rest }) => (
    <Box as="span" display="block" fontSize={0} {...rest}>
      {label}
    </Box>
  )
);

export const Tooltip: React.FC<
  TippyProps & { label?: TippyProps["content"]; labelProps?: BoxProps }
> = memo(({ label, labelProps = {}, children, ...rest }) => {
  if (!label) return <>{children}</>;
  return (
    <Tippy
      content={<Content label={label} {...labelProps} />}
      trigger="mouseenter"
      hideOnClick={undefined}
      {...rest}
    >
      {children}
    </Tippy>
  );
});
```